### PR TITLE
[FIX] mobile: container details to article details

### PR DESCRIPTION
**Pull request by invpuppet**
fix route (become 'get' routes):
- /api/container/
- /api/container/articleslist/
- /api/article/

add tests for (front and back) :
- container details
- article list
- article list data
- article details

fix theme colors of the aforementioned pages

### CODE REVIEW

**Review by 31Nathan — COMMENTED** [review]
Overall, seems ok, just need to know why the article.dart was deleted

File: Application/mobile/app_code/test/article_details_test.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          ),
+        ),
+      );
+
+      Finder titleData = find.byKey(const Key('article-details_title'));
+
+      expect(titleData, findsOneWidget);
+    }
+  );
+}
```
Comment (tbellicha) [inline Application/mobile/app_code/test/article_details_test.dart:38]: Missing new line

File: Application/mobile/app_code/test/article_list_test.dart [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            home: ArticleListPage(containerId: ''),
+          ),
+        ),
+      );
+
+      Finder articleCard = find.byKey(const Key('articles-list_card'));
+      expect(articleCard, findsNothing);
+    }
+  );
+}
```
Comment (tbellicha) [inline Application/mobile/app_code/test/article_list_test.dart:37]: Missing new line

File: Application/server/server.js [hunk 3]
```diff
@@ -471,23 +479,18 @@ app.get('/api/container/listall',
   }
 )
 
-app.post('/api/container/details',
+app.get('/api/container/:containerId',
   passport.authenticate('jwt', { session: false }), async (req, res) => {
     try {
+      console.log("container/details")
```
Comment (tbellicha) [inline Application/server/server.js:485]: Remove useless log

File: Application/server/server.js [hunk 4]
```diff
@@ -508,14 +511,18 @@ app.post('/api/container/details',
     }
   })
 
-app.post('/api/container/articleslist',
+app.get('/api/container/articleslist/:containerId',
   passport.authenticate('jwt', { session: false }), async (req, res) => {
     try {
-      if (!req.body.containerId || req.body.containerId === '') {
+      console.log("container/articleList")
```
Comment (tbellicha) [inline Application/server/server.js:517]: Remove useless log

**Comment by invpuppet** [comment]
image
image
image

File: Application/server/server.js [hunk 5]
```diff
@@ -528,8 +535,9 @@ app.post('/api/container/articleslist',
           }
         },
       })
+      console.log(container);
       if (!container) {
-        return res.status(401).json("containerlist not found")
+        return res.status(401).json("itemList not found")
       } else if (!container.items || container.items.length === 0) {
         return res.status(204).json({ message: 'Container doen\'t have items' })
```
Comment (tbellicha) [inline Application/server/server.js:542]: doesn't*

File: Application/server/server.js [hunk 6]
```diff
@@ -639,7 +647,7 @@ app.get('/api/rent',
       console.error(err.message)
       res.status(401).send('An error occurred')
```
Comment (tbellicha) [inline Application/server/server.js:648]: use return, 2 lines above too

File: Application/server/server.js [hunk 7]
```diff
@@ -639,7 +647,7 @@ app.get('/api/rent',
...
     }
-})
+  })
```
Comment (tbellicha) [inline Application/server/server.js:650]: why these spaces ?
LIne 627 too

File: Application/server/tests/apiArticle.test.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+          const res = await request('http://localhost:8080')
+            .get(`/api/article/${articleId}`)
+            .set('Authorization', `Bearer article token`)
+          expect(res.statusCode).toBe(401)
+        }
+      ],
+      done
+    )
+  })
+})
```
Comment (tbellicha) [inline Application/server/tests/apiArticle.test.js:80]: Missing new line

File: Application/server/tests/apiContainer.test.js [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+            const res = await request('http://localhost:8080')
+              .get(`/api/container/articleslist/${containerId[0]}`)
+              .set('Authorization', `Bearer invalidToken`)
+            expect(res.statusCode).toBe(401)
+          }
+        ],
+        done
+      )
+    })
+})
```
Comment (tbellicha) [inline Application/server/tests/apiContainer.test.js:148]: Missing new line

**Review by tbellicha — CHANGES_REQUESTED** [review]

**Pull request merged**